lib/active_merchant/billing/gateways/elavon.rb

```diff
@@ -370,6 +373,15 @@ def merchant_initiated_unscheduled(options)
         return 'Y' if options.dig(:stored_credential, :initiator) == 'merchant' && options.dig(:stored_credential, :reason_type) == 'unscheduled'
       end
 
+      def recurring_txn(options)
+        return options[:merchant_initiated_unscheduled] if options[:merchant_initiated_unscheduled]
+        return 'Y' if options.dig(:stored_credential, :initiator) == 'merchant' && options.dig(:stored_credential, :reason_type) == 'recurring'
+        return options[:ssl_get_token] if options[:ssl_get_token]
+        return 'Y' if options.dig(:stored_credential, :reason_type) == 'recurring' && options.dig(:store_credential, :initial_transaction) == true
+        return options[:ssl_add_token] if options[:ssl_add_token]
+        return 'Y' if options.dig(:stored_credential, :reason_type) == 'recurring' && options.dig(:store_credential, :initial_transaction) == true
```

Comment:
I think this last `return` here is a duplicate.

I do think we can clean this up a little. It looks like the value we want to return will be 'Y' or we don't want to do anything. We can combine `377-381` into one conditional check, perhaps like this:

```ruby
  if options.dig(:stored_credential, :reason_type) == 'recurring' && (options.dig(:stored_credential, :initial_transaction) == true || options.dig(:stored_credential, :initiator) == 'merchant')
      return 'Y'
  elsif [options.dig(:merchant_initiated_unscheduled), options.dig(:ssl_get_token), options.dig(:ssl_add_token)].include?('Y')
      return 'Y'
  end
```

For the first check, if we know that `reason_type` is `recurring` AND one of the other 2 conditions is "true", then we can return 'Y'. Else, check if one of the other 3 fields (`ssl_add_token`, `ssl_get_token`, or `merchant_initiated_unscheduled`) includes 'Y' (which I assume is what we expect to be passed there).

I know these stored credential fields can get a bit hairy, but I think this is a bit more readable. Feel free to take it and play around with it, I only tested it against your test without trying to throw a wrench into it.